client/app/bundles/course/experience-points/components/ExperiencePointsTable.tsx

```diff
@@ -0,0 +1,98 @@
+import { Dispatch, FC, memo, SetStateAction, useEffect } from 'react';
+import { defineMessages } from 'react-intl';
+import { TableBody, TableCell, TableHead } from '@mui/material';
+import equal from 'fast-deep-equal';
+
+import { setNotification } from 'lib/actions';
+import TableContainer from 'lib/components/core/layouts/TableContainer';
+import LoadingIndicator from 'lib/components/core/LoadingIndicator';
+import { useAppDispatch, useAppSelector } from 'lib/hooks/store';
+import useTranslation from 'lib/hooks/useTranslation';
+import tableTranslations from 'lib/translations/table';
+
+import {
+  fetchAllExperiencePointsRecord,
+  fetchUserExperiencePointsRecord,
+} from '../operations';
+import { getAllExpPointsRecordsEntities } from '../selectors';
+
+import ExperiencePointsTableRow from './ExperiencePointsTableRow';
+
+interface Props {
+  isDownloading?: boolean;
+  studentId?: number;
+  pageNum: number;
+  isStudentPage?: boolean;
+  isLoading: boolean;
+  setIsLoading: Dispatch<SetStateAction<boolean>>;
+}
+
+const translations = defineMessages({
+  fetchRecordsFailure: {
+    id: 'course.experiencePoints.fetchRecordsFailure',
+    defaultMessage: 'Failed to fetch records',
+  },
+});
+
+const ExperiencePointsTable: FC<Props> = (props) => {
+  const {
+    studentId,
+    pageNum,
+    isStudentPage,
+    isLoading,
+    setIsLoading,
+    isDownloading,
+  } = props;
+  const dispatch = useAppDispatch();
+
+  const { t } = useTranslation();
+
+  const records = useAppSelector(getAllExpPointsRecordsEntities);
+
+  const fetchExperiencePoints = isStudentPage
+    ? fetchUserExperiencePointsRecord(studentId!, pageNum)
+    : fetchAllExperiencePointsRecord(studentId, pageNum);
+
+  useEffect(() => {
```

Comment:
The responsibility of `ExperiencePointsTable` is merely to display the data. The data fetching should not be mixed. There needs to be a separation of responsibility between components here.

Now as you have seen, because of that, you need to pass isStudentPage and check for studentId. You have if else and you are forcing studentId to be non-undefined in line 53 above.

I know it is based on the old-code but in this case, the old code is not correct.

Reply:
I have moved this component accordingly
however, this indicator of whether the component is called in student page or not is still important, since the table displayed at `course/{course_id}/experience_points_records` and `course/{course_id}/users/{user_id}/experience_points_records` has one subtle differences (in the latter, we don't need to display the student's name in each row, but we need to do that in the former one)